application/src/main/java/org/togetherjava/tjbot/commands/help/BookmarkPaginatorInteractor.java

```diff
@@ -0,0 +1,479 @@
+            .toList();
+    }
+
+    /**
+     * Calculates the last page index for displaying all bookmarks.
+     *
+     * @param bookmarks The bookmarks list
+     * @return The index of the last page
+     */
+    private static int getLastPageIndex(List<BookmarksRecord> bookmarks) {
+        int highestBookmarkIndex = bookmarks.size() - 1;
+
+        return Math.floorDiv(highestBookmarkIndex, ENTRIES_PER_PAGE);
+    }
+
+    /**
+     * Makes sure that the page index never is negative or larger than the index of the last page.
+     *
+     * @param bookmarks The bookmarks list
+     * @param pageIndex The page possibly invalid index
+     * @return A valid page index
+     */
+    private static int ensurePageIndex(List<BookmarksRecord> bookmarks, int pageIndex) {
+        // If the bookmarks list is empty the maxBookmarkIndex would be -1. This causes the
+        // maxPageIndex to also return -1, resulting in other errors.
+        if (bookmarks.isEmpty()) {
+            return 0;
+        }
+
+        int maxBookmarkIndex = bookmarks.size() - 1;
+        int maxPageIndex = Math.floorDiv(maxBookmarkIndex, ENTRIES_PER_PAGE);
+
+        pageIndex = Math.max(pageIndex, 0);
+        pageIndex = Math.min(pageIndex, maxPageIndex);
+
+        return pageIndex;
+    }
+
+    private static ComponentArguments parseStringArgs(List<String> stringArgs) {
+        return stringArgs.get(0).equals(PaginationType.REMOVE.name())
+                ? RemoveComponentArguments.fromStringArgs(stringArgs)
+                : ComponentArguments.fromStringArgs(stringArgs);
+    }
+
+    private static List<String> parseListToStringList(List<?> list) {
+        return list.stream().map(String::valueOf).toList();
+    }
+
+    private enum PaginationType {
+        LIST,
+        REMOVE
+    }
+
+
+    private record PageEntry(int bookmarkNumber, BookmarksRecord bookmark) {
+    }
+
+
+    private static class ComponentArguments implements Arraylizable<String> {
+
+        final PaginationType paginationType;
+        final String componentName;
+        int currentPageIndex;
+
+        public ComponentArguments(PaginationType paginationType, String componentName,
+                int currentPageIndex) {
+            this.paginationType = paginationType;
+            this.currentPageIndex = currentPageIndex;
+            this.componentName = componentName;
+        }
+
+        @Override
+        public String[] toArray() {
+            return new String[] {paginationType.name(), componentName,
+                    String.valueOf(currentPageIndex)};
+        }
+
+        public static ComponentArguments fromStringArgs(List<String> stringArgs) {
+            PaginationType paginationType = PaginationType.valueOf(stringArgs.get(0));
+            String componentName = stringArgs.get(1);
+            int currentPageIndex = Integer.parseInt(stringArgs.get(2));
+
+            return new ComponentArguments(paginationType, componentName, currentPageIndex);
+        }
```

Comment:
ComponentArguments and RemoveComponentArguments have these methods in reversed order